doc/developer-notes.md

```diff
@@ -839,6 +840,37 @@ namespace {
 #endif // BITCOIN_FOO_BAR_H
 ```
 
+Construct On First Use
+------------------------
+
+It is not possible to control the order in which non-local static objects in *different* translation units are initialized.
+This is the so-called "static initialization order problem". It could cause the use of an object:
+- before it is constructed (on program startup)
+- after it is destructed (at program exit)
+
+Both cases lead to undefined behavior.
+
+To prevent the static initialization order problem, use the Construct On First Use:
+
+```C++
+BCLog::Logger& LogInstance()
+{
+    // See doc/developer-notes.md#construct-on-first-use
```

Comment:
comment seems unneeded?

Reply:
It seems developers need multi-line comments to describe rationale of using  COFU :)
This oneliner is just a pointer to docs with detailed explanation and pitfalls (like not calling a destructor). I'd prefer keep it as is.

Reply:
Yes, in the codebase it's fine but here it's a pointer to itself :)

Reply:
As it is an "example", the presence of the comment line seems reasonable, no?